### PR TITLE
APS-2719 Recover the agent when a connection to LDAP server fails

### PR DESCRIPTION
**Pull request by fschuch**
Porting the patch applyed to `main` on #66 to the release brach.

### CODE REVIEW

**Review by matheushent — APPROVED** [review]

File: cluster_agent/utils/logging.py [hunk 1]
```diff
@@ -24,3 +25,32 @@ def log_error(params: DoExceptParams):
...
+        while frame.f_code.co_filename == logging.__file__:
+            frame = frame.f_back
+            depth += 1
+
+        logger.opt(depth=depth, exception=record.exc_info).log(
+            level, record.getMessage()
+        )
+
+
+logging.basicConfig(handlers=[InterceptHandler()], level=0)
```
Comment (julianaklulo) [inline cluster_agent/utils/logging.py:56]: Why is this needed?
